src/deprecations.jl

```diff
@@ -0,0 +1,50 @@
+# have to import this or @deprecate doesn't work
+import Base: getindex, setindex!, delete!, haskey, get, copy!, getproperty, setproperty!
+
+@deprecate(ImageMeta(data::AbstractArray{T,N}, props::AbstractDict{String,Any}) where {T,N},
+           ImageMeta(data, to_dict(props)))
+
+@deprecate(ImageMeta(data::AbstractArray{T,N}, props::Dict{String,Any}) where {T,N},
+           ImageMeta(data, to_dict(props)))
+
+@deprecate(ImageMeta(data::AbstractArray, props::Dict{<:AbstractString}),
+           ImageMeta(data, to_dict(props)))
+
+@deprecate(copy!(imgdest::ImageMeta, imgsrc::ImageMeta, prop1::AbstractString, props::AbstractString...),
+           copy!(imgdest, imgsrc, Symbol(prop1), Symbol.(props)...))
+
+@deprecate(delete!(img::ImageMeta, propname::AbstractString),
+           delete!(img, Symbol(propname)))
+
+@deprecate(haskey(img::ImageMeta, k::AbstractString),
+           hasproperty(img, Symbol(k)))
+
+@deprecate(hasproperty(img::ImageMeta, k::AbstractString),
+           hasproperty(img, Symbol(k)))
+
+@deprecate(get(img::ImageMeta, k::AbstractString, default),
+           get(img, Symbol(k), default))
+
+@deprecate(getindex(img::ImageMeta, propname::AbstractString),
+           getproperty(img, Symbol(propname)))
+
+@deprecate(getproperty(img::ImageMeta, propname::AbstractString),
+           getproperty(img, Symbol(propname)))
+
+@deprecate(setproperty!(img::ImageMeta, propname::AbstractString, val),
+           setproperty!(img::ImageMeta, Symbol(propname), val))
+
+@deprecate(setindex!(img::ImageMeta, X, propname::AbstractString),
+           setproperty!(img, Symbol(propname), X))
+
+@deprecate(getproperty_data(img), data(img))
```

Comment:
I see the point here, but I'm not sure if this is appropriate because I think the following depwarn isn't so informative:

```julia
┌ Warning: `getproperty_data(img)` is deprecated, use `data(img)` instead.
│   caller = getproperty(::ImageMeta{Float64,2,AxisArray{Float64,2,Array{Float64,2},Tuple{Axis{:y,Base.OneTo{Int64}},Axis{:x,Base.OneTo{Int64}}}},Dict{Symbol,Any}}, ::Symbol) at ImageMetadata.jl:140
```

So you might need to manually construct a depwarn instead of using `@deprecate`, a live [example](https://github.com/zygmuntszpak/ImageBinarization.jl/blob/e6b3b13279196544e815c821e45e2bef37c18376/src/deprecations.jl#L3-L8 for an example)

Reply:
Thanks. I wasn't sure how to do this